address a list of failing tests for chakra

jianchun opened the pull request:
This PR contains 2 fixes in chakrashim and a list of test patches.

Reduced outstanding test failures on node-chakracore from ~39 to ~16.

Diffing with my chakracore-master_20160406 log:

```
InputObject                                                                 SideIndicator
-----------                                                                 -------------
addons/load-long-path/test                                                  =>
addons/repl-domain-abort/test                                               <=
message/vm_display_runtime_error                                            <=
message/vm_dont_display_runtime_error                                       <=
parallel/test-async-wrap-throw-from-callback                                <=
parallel/test-buffer                                                        <=
parallel/test-buffer-alloc                                                  <=
parallel/test-buffer-concat                                                 <=
parallel/test-buffer-includes                                               <=
parallel/test-buffer-indexof                                                <=
parallel/test-child-process-flush-stdio                                     <=
parallel/test-domain-abort-on-uncaught                                      <=
parallel/test-domain-no-error-handler-abort-on-uncaught                     <=
parallel/test-domain-throw-error-then-throw-from-uncaught-exception-handler <=
parallel/test-domain-with-abort-on-uncaught-exception                       <=
parallel/test-error-reporting                                               <=
parallel/test-fs-readfile-tostring-fail                                     <=
parallel/test-intl                                                          <=
parallel/test-regress-GH-io-1068                                            <=
parallel/test-repl                                                          <=
parallel/test-repl-require                                                  <=
parallel/test-repl-syntax-error-stack                                       <=
parallel/test-util-decorate-error-stack                                     <=
parallel/test-util-internal                                                 <=
sequential/test-regress-GH-746                                              <=
```

Commit:
chakrashim: let caller TryCatch record exception

It's possible a caller uses TryCatch, call an external function which
throws an exception, then continue to make other calls to script engine.
The exception is supposed to be caught by the TryCatch.

Fixes `test-async-wrap-throw-from-callback` on node-chakracore.

PR-URL: https://github.com/nodejs/node-chakracore/pull/54
Reviewed-By: Sandeep Agarwal <Agarwal.Sandeep@microsoft.com>

Commit:
test: patch tests for chakra

Patch a list of failing tests for chakra.
- skip tests dependent on v8-option --abort-on-uncaught-exception.
- skip a test-repl sequence due to lack of chakracore support.
  Filed a chakracore issue to track.
- skip tests dependent on `new Uint8Array(NaN)`
- patch tests dependent on `new Uint8Array(floatValue)`
- patch tests dependent on Error message/script/source
- patch tests dependent on String limit
- patch tests dependent on Date format output
- add chakracore outputs for 2 message tests.

PR-URL: https://github.com/nodejs/node-chakracore/pull/54
Reviewed-By: Sandeep Agarwal <Agarwal.Sandeep@microsoft.com>

kunalspathak commented on test/parallel/test-intl.js:
```diff
@@ -60,14 +60,21 @@ if (!haveIntl) {
-  assert.equal(localeString, 'Jan 70');
+  assert.equal(localeString, common.engineSpecificMessage({
+    v8: 'Jan 70',
+    chakracore: '\u200EJan\u200E \u200E70'
```
I noticed this failure earlier and didn't patch intentionally for chakra because it is bug in chakracore or `globalization.dll`. Till we confirm, I would leave this test case from patching. I could even repro similar failure with `ch.exe`.

jianchun replied:
This issue is known and being discussed at ChakraCore and TC39:

https://github.com/Microsoft/ChakraCore/issues/599
https://github.com/tc39/ecma402/issues/28

Patching it enables rest of the test to run and gives us more test coverage.

kunalspathak replied:
Sounds good to me then.

Sent from Outlook Mobilehttps://aka.ms/blhgte

On Sat, Apr 9, 2016 at 7:38 AM -0700, "Jianchun Xu" <notifications@github.com<mailto:notifications@github.com>> wrote:

In test/parallel/test-intl.jshttps://github.com/nodejs/node-chakracore/pull/54#discussion_r59116119:

> @@ -60,14 +60,21 @@ if (!haveIntl) {
> 
>    // Check with toLocaleString
>    var localeString = dtf.format(date0);
> -  assert.equal(localeString, 'Jan 70');
> -  assert.equal(localeString, common.engineSpecificMessage({
> -    v8: 'Jan 70',
> -    chakracore: '\u200EJan\u200E \u200E70'

This issue is known and being discussed at ChakraCore and TC39:

Microsoft/ChakraCore#599https://github.com/Microsoft/ChakraCore/issues/599
tc39/ecma402#28https://github.com/tc39/ecma402/issues/28

Patching it enables rest of the test to run and gives us more test coverage.

## 

You are receiving this because you commented.
Reply to this email directly or view it on GitHubhttps://github.com/nodejs/node-chakracore/pull/54/files/2fc922aaa3e31e99fca87e1b46817d41c48986f5..c042de3faa269497ca0b8406706476dd717ff0e4#r59116119

Commit:
test: fix wrong python script indentation

A previous chakra-specific customization of `test/message/testcfg.py`
introduced wrong indentation (42bc2848a89a9f9ee145f9fdfe057465c410b2d9).
This causes all `.chakracore.out` message tests be skipped instead of
customized.

Fixed the wrong indentation.

PR-URL: https://github.com/nodejs/node-chakracore/pull/54
Reviewed-By: Sandeep Agarwal <Agarwal.Sandeep@microsoft.com>

Commit:
test: update 5 node-chakracore message test baselines

PR-URL: https://github.com/nodejs/node-chakracore/pull/54
Reviewed-By: Sandeep Agarwal <Agarwal.Sandeep@microsoft.com>

Commit:
chakrashim: correct function callback context

A native function callback is expected to be invoked in its creation
context. Enter the callee context before invoking native callback.

Fixes several 'repl` related tests:
- parallel/test-repl-require
- parallel/test-repl-syntax-error-stack
- addons/repl-domain-abort/test

PR-URL: https://github.com/nodejs/node-chakracore/pull/54
Reviewed-By: Sandeep Agarwal <Agarwal.Sandeep@microsoft.com>
